src/guidellm/utils/messaging.py

```diff
@@ -0,0 +1,1056 @@
+        pydantic_models: list[type[BaseModel]] | None = None,
+    ):
+        """
+        Start asynchronous message processing tasks with buffering.
+
+        :param send_items: Optional collection of items to send during processing
+        :param receive_callback: Optional callback for processing received messages
+        :param stop_events: External events that trigger messaging shutdown
+        :param send_stop_events: Events that trigger send task shutdown
+        :param receive_stop_events: Events that trigger receive task shutdown
+        :param pydantic_models: Optional list of Pydantic models for serialization
+        """
+        self.running = True
+        self.stopped_event = ThreadingEvent()
+        self.shutdown_event = ThreadingEvent()
+        self.buffer_send_queue = culsans.Queue[SendMessageT]()
+        self.buffer_receive_queue = culsans.Queue[ReceiveMessageT]()
+
+        message_encoding = MessageEncoding(
+            serialization=self.serialization,
+            encoding=self.encoding,
+            pydantic_models=pydantic_models,
+        )
+        if send_stop_events is None:
+            send_stop_events = []
+        if receive_stop_events is None:
+            receive_stop_events = []
+        if stop_events:
+            send_stop_events.extend(stop_events)
+            receive_stop_events.extend(stop_events)
+
+        self.send_task = asyncio.create_task(
+            self.send_messages_task(
+                message_encoding=message_encoding,
+                stop_events=send_stop_events,
+                send_items=send_items,
+            )
+        )
+        self.receive_task = asyncio.create_task(
+            self.receive_messages_task(
+                message_encoding=message_encoding,
+                stop_events=receive_stop_events,
+                receive_callback=receive_callback,
+            )
+        )
+
+    async def stop(self):
+        """
+        Stop message processing tasks and clean up resources.
+        """
+        self.shutdown_event.set()
+        with contextlib.suppress(asyncio.CancelledError):
+            await asyncio.gather(
+                self.send_task, self.receive_task, return_exceptions=True
+            )
+        self.send_task = None
+        self.receive_task = None
+        await self.buffer_send_queue.aclose()
+        await self.buffer_receive_queue.aclose()
+        self.buffer_send_queue = None
+        self.buffer_receive_queue = None
+        self.stopped_event = None
+        self.shutdown_event = None
+        self.running = False
+
+    async def get(self, timeout: float | None = None) -> ReceiveMessageT:
+        """
+        Retrieve message from receive buffer with optional timeout.
+
+        :param timeout: Maximum time to wait for a message
+        :return: Decoded message from the receive buffer
+        """
+        return await asyncio.wait_for(
+            self.buffer_receive_queue.async_get(), timeout=timeout
+        )
```

Comment:
Does this function return a `TimeoutError` when it times out? If so I think it would be beneficial to document that on all of these functions that have a timeout.